### DISCUSSION TITLE
Select filter with relationship: Cannot display name from 'getNameAttribute()'

### DISCUSSION THREAD

**Discussion by valpuia** [filamentphp/filament#6580] [Help]
I have a translation table where I can add dynamic country data with some other language, If they change the locale then based on the `locale+language_id+session` I changed the `name attribute` for particular language like below

```
// Country.php
public function translation(): MorphMany
{
    return $this->morphMany(Translation::class, 'translationable');
}

public function getNameAttribute()
{
    return $this->translation()->where('language_id', getLocaleId())->value('translation') ?: $this->attributes['name'];
}

// Currency.php // CurrencyResource is where I try to display the country data
public function country(): BelongsTo
{
    return $this->belongsTo(Country::class);
}

// helpers.php
function getLocaleId()
{
    return \App\Models\Language::where('locale', session('locale'))->value('id');
}
```

Using this I can display the `name` for particular language in table without writing extra code.

I have to use this logic inside `form select` and I can achieve using this `->getOptionLabelFromRecordUsing(fn (Model $record) => $record->name)`

And again I try to use this logic in `Select filter` using this [https://filamentphp.com/docs/2.x/tables/filters#relationship-select-filters](https://filamentphp.com/docs/2.x/tables/filters#relationship-select-filters)

```
Tables\Filters\SelectFilter::make('country_id')
     ->label(__('message.country'))
     ->relationship('country', 'name', fn (Builder $query) => $query->active()),
```

But here I cannot use my `name attribute logic` to change country name base on locale..

I am doing wrong or is this a bug,
or is there any other solution to display the label like in `form select` `getOptionLabelFromRecordUsing()`

**Comment by danharrin** [comment]
You could override the `options()` array on the filter to map each country to `$country->name` in a similar way. The reason that accessors don't work is because we use Laravel's `pluck()` to avoid loading all models into memory when we dont need to.
